lib/TH/THTensorApply.h

```diff
@@ -34,70 +36,131 @@
   else \
   { \
     TENSOR1##_data = TENSOR1->storage->data+TENSOR1->storageOffset; \
-    for(TENSOR1##_dim = TENSOR1->nDimension-1; TENSOR1##_dim >= 0; TENSOR1##_dim--) \
-    { \
-      if(TENSOR1->size[TENSOR1##_dim] != 1) \
-        break; \
-    } \
-    TENSOR1##_stride = (TENSOR1##_dim == -1 ? 0 : TENSOR1->stride[TENSOR1##_dim]); \
     TENSOR1##_size = 1; \
-    for(TENSOR1##_dim = TENSOR1->nDimension-1; TENSOR1##_dim >= 0; TENSOR1##_dim--) \
-    { \
-      if(TENSOR1->size[TENSOR1##_dim] != 1) \
+    TENSOR1##_stride = 1; \
+    for(TENSOR1##_i = TENSOR1->nDimension-1; TENSOR1##_i >= 0; TENSOR1##_i--) { \
```

Comment:
Yes. During the compilation period, torch generates different types of `isCoutiguous` function. That is to say, `double`, `float` or `byte` are appended before `isContiguous`. However, indeed the function is independent of types. Therefore, we cannot use `isContiguous` in `THTensorApply2` or `THTensorApply3` when array types are different